File: 3392/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(countSubarrays([]int{1,2,1,4,1})) // 1
	fmt.Println(countSubarrays([]int{0,0,0,0})) // 2
	fmt.Println(countSubarrays([]int{-1,-4,-1,4})) // 1
}

func countSubarrays(nums []int) int {
	n := len(nums)
	ans := 0
	for i := 2; i < n; i++ {
		if (nums[i] + nums[i-2])*2 == nums[i-1] {
			ans++
		}
	}
	return ans
}

File: 3392/main.rs
struct Solution{}

impl Solution {
    pub fn count_subarrays(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut ans = 0;
        for i in 1..n-1 {
            if nums[i] == (nums[i-1] + nums[i+1]) * 2 {
                ans += 1;
            }
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::count_subarrays(vec![-1,-4,-1,4])) // 1
}